packages/apputils/src/bidiSTT.ts

```diff
@@ -0,0 +1,626 @@
+/**
+ * The bidi event raised when an HTML element's content is changed  
+ */
+var event: Event = null;
+  
+export
+namespace bidiSTT {
+  export
+  /**
+   * Attach a bidi event to an HTML element 
+   * When bidi event is raised the relevant handler is activated to parse and display the content of the element in the relevant structure
+   * Structures can be file paths, bread crumbs, e-mails, formulas etc. Currently only file paths are supported.   
+   * 
+   * @param element The HTML element to attach the event to
+   * @param type    The handler type - The type of structured text required. 
+   */  
+  function attachElement(element: HTMLElement, type: string) {
```

Comment:
Thanks for the formatting changes and the comments, I understand much better now :smile:.  I think `attachElement` is confusing in this context because we use the term `attach` in Phosphor to mean that a widget's node is added to the DOM.  Perhaps `applyBidi()` as was used in the corresponding notebook [PR](https://github.com/jupyter/notebook/pull/2357)?

Reply:
Sure, it makes sense (I applied the changes in the main thread).

Reply:
Thanks!  I like this approach overall.  Perhaps we should use the bidi file name handler in the [file listing](https://github.com/jupyterlab/jupyterlab/blob/793dd5ff16ced7329604b9ce13ed3413f32f1824/packages/filebrowser/src/listing.ts#L1681) as a first step since we are blocked on the document model refactor?

Reply:
@blink1073 Sorry, I don’t quite follow you.
Can you please explain what you mean by 'use the bidi file name handler in the file listing'
Do you mean – apply bidi code on file and folder names displayed in the file browser?

Reply:
Yep!